ClientApp/viking-vault/src/UserDashboard/DownloadTypeForm.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { Redirect } from 'react-router-dom';
+import downloadIcon from '../Resources/images/download-icon.png'
+import { constants } from "../Resources/Constants";
+
+const API_URL = `${constants.baseUrl}pdfgenerator`;
+const LOGIN_ROUTE ="/login";
+
+class DownloadDropdown extends React.Component {
+    state = {
+      isOpen: false,
+      timeFilter: "",
+      redirect: false
+    };
+  
+    toggleOpen = () => this.setState({ isOpen: !this.state.isOpen });
+  
+    generatePDF = () => {
+        let token = sessionStorage.getItem("Authentication-Token");
+        if(token === null){
+            this.setState({
+                redirect: true
+            })
+        }
+        else {
+            var pdfURL = API_URL + "/"+ this.state.timeFilter;
+            fetch(pdfURL, {
+                    method: "GET",
+                    headers: {
+                        "x-access-token":  token.toString()
+                    }
+                })
+                .then((response) => response.blob())
+                .then((blob) => {
+                    if(blob != null) {
+                    const url = window.URL.createObjectURL(new Blob([blob]));
+                    const link = document.createElement('a');
+                    link.href = url;
+                    link.setAttribute('download', `Viking Vault Report.pdf`);
+                    link.click(); }
+                })
+            }
+        }
+
+    handlePDFRequestPer = (timeFilter: string) => {
+        this.setState({
+            timeFilter: timeFilter,
+            isOpen: !this.state.isOpen
+        }, this.generatePDF);
+        return null;
```

Comment:
Kind of redundant to return `null` here, if a method is void, just don't return return (and don't look back 😅), eventually use the typescript type of `void` as the function's return type to explicitly tell that this doesn't return any data

Reply:
I removed it, thankies.